superset/views/core.py

```diff
@@ -1380,10 +1380,17 @@ def created_dashboards(  # pylint: disable=no-self-use
     def user_slices(  # pylint: disable=no-self-use
         self, user_id: Optional[int] = None
     ) -> FlaskResponse:
-        """List of slices a user created, or faved"""
+        """List of slices a user owns or faved"""
         if not user_id:
             user_id = g.user.id
         FavStar = models.FavStar
+
```

Comment:
We should make sure `<int:user_id>` is the current user. 
The best solution should be to just expose `/user_slices/`, @john-bodley , @etr2460 can we remove this?

Reply:
I agree, and I was surprised by the signature. There's a few similar endpoints under `views`, perhaps tackle those in one go in a separate PR?

Reply:
sounds good!